homeassistant/components/media_player/enigma.py

```diff
@@ -0,0 +1,164 @@
+"""
+homeassistant.components.media_player.enigma
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Beta implementation of the Enigma2 media player.
+Currently only shows whats playing on the box.
+
+You should have a recent version of OpenWebIf installed
+on your E2 box.
+
+There is no support for username/password authentication
+at this time.
+
+Configuration:
+
+To use the media_player you will need to add something like the
+following to your config/configuration.yaml
+
+media_player:
+    platform: enigma
+    name: Vu Duo2
+    host: 192.168.1.26
+    port: 80
+
+Variables:
+
+host
+*Required
+This is the IP address of your Enigma2 box. Example: 192.168.1.32
+
+port
+*Optional
+The port your Enigma2 box uses, defaults to 80. Example: 8080
+
+name
+*Optional
+The name to use when displaying this Enigma2 switch instance.
+
+"""
+from homeassistant.components.media_player import (
+    MediaPlayerDevice, STATE_NO_APP, ATTR_MEDIA_STATE,
+    ATTR_MEDIA_CONTENT_ID, ATTR_MEDIA_TITLE, ATTR_MEDIA_DURATION,
+    ATTR_MEDIA_VOLUME, MEDIA_STATE_PLAYING, MEDIA_STATE_STOPPED)
+
+import logging
+import openwebif.api
+_LOGGING = logging.getLogger(__name__)
+
+# pylint: disable=unused-argument
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Sets up the enigma media player platform. """
+
+    host = config.get('host', None)
+    port = config.get('port', "80")
+    name = config.get('name', "Enigma2")
+
+    if not host:
+        _LOGGING.error('Missing config variable-host')
+        return False
+
+    e2_box = openwebif.api.Client(host, port=port)
+
+    add_devices([
+        EnigmaMediaPlayer(name, e2_box)
+    ])
+
+
+class EnigmaMediaPlayer(MediaPlayerDevice):
+
+    """ An Enigma2 media player. """
+
+    def __init__(self, name, e2_box):
+        self._name = name
+        self._e2_box = e2_box
+        self.state_attr = {ATTR_MEDIA_STATE: MEDIA_STATE_STOPPED}
+        # self.update()
+
+        self.is_playing = False
+        self.media_title = None
+        self.volume = 1.0
+        self.channel_title = None
+
+    @property
+    def should_poll(self):
+        """ Need to refresh ourselves. """
+        return True
+
+    @property
+    def name(self):
+        """ Returns the name of the device. """
+        return self._name
+
+    @property
+    def state(self):
+        """ Returns the state of the device. """
+        return STATE_NO_APP if self.is_playing is False else self.channel_title
+
+    @property
+    def state_attributes(self):
+        """ Returns the state attributes. """
+
+        return self.state_attr
```

Comment:
Ca we move the logic to generate the state attributes to this method instead of the update method?